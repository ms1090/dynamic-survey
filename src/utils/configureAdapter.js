import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
// configure adapter to use enzyme in unit test cases
configure({ adapter: new Adapter() });
