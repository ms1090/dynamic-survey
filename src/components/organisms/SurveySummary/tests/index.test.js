import React from 'react';
import { shallow } from 'enzyme';
import { SurveySummaryVanilla } from '..';
import toJson from 'enzyme-to-json';
import 'utils/configureAdapter';

describe('SurveySummaryVanilla Component', () => {
  const questions = {
    allIds: ['ques1', 'ques2'],
    byId: {
      ques1: {
        id: 'ques1',
        text: 'What do you think about the use of dummies?',
        options: [
          "It's fine- whatever works, right?",
          "I don't mind them but never used much.",
          'I preferred to not use the dummy as much as possible.',
          'I would never use a dummy.',
        ],
      },
      ques2: {
        id: 'ques2',
        text: 'Can you tell us why you have this view?',
      },
    },
    currQuesIndex: 0,
  };
  const restartSurveyMock = jest.fn();
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<SurveySummaryVanilla
      questions={questions}
      restartSurvey={restartSurveyMock}
    />);
  });
  test('should render correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test('should call restartSurvey on "restart survey" button click', () => {
    const buttonEle = wrapper.find('Button');
    expect(buttonEle.exists()).toBe(true);
    buttonEle.simulate('click');
    expect(restartSurveyMock.mock.calls.length).toBe(1);
  });
});
