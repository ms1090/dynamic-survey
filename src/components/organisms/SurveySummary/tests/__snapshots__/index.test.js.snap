// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SurveySummaryVanilla Component should render correctly 1`] = `
<section>
  <QuestionResponse
    index={1}
    key="ques1"
    question={
      Object {
        "id": "ques1",
        "options": Array [
          "It's fine- whatever works, right?",
          "I don't mind them but never used much.",
          "I preferred to not use the dummy as much as possible.",
          "I would never use a dummy.",
        ],
        "text": "What do you think about the use of dummies?",
      }
    }
  />
  <QuestionResponse
    index={2}
    key="ques2"
    question={
      Object {
        "id": "ques2",
        "text": "Can you tell us why you have this view?",
      }
    }
  />
  <Divider />
  <Button
    className=""
    onClick={[Function]}
    primary={false}
    secondary={false}
  >
    RESTART SURVEY
  </Button>
</section>
`;
